examples/MNIST_classification.ipynb

```diff
@@ -0,0 +1,598 @@
+    "from gtda.homology import CubicalPersistence\n",
+    "from gtda.diagrams import ForgetDimension, Amplitude, Scaler, PersistenceEntropy, BettiCurve, PersistenceLandscape, HeatKernel, Silhouette\n",
+    "from gtda.plotting import plot_heatmap, plot_betti_curves, plot_diagram\n",
+    "from sklearn.pipeline import Pipeline, make_pipeline, FeatureUnion, make_union\n",
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "from sklearn.model_selection import GridSearchCV\n",
+    "from sklearn.metrics import classification_report\n",
+    "import numpy as np\n",
+    "\n",
+    "# This would be quite nice but is available with sklearn >= 0.23\n",
+    "# from sklearn import set_config\n",
+    "#set_config(display='diagram') "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading the MNIST dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.datasets import fetch_openml\n",
+    "\n",
+    "(X, y) = fetch_openml(data_id=554, return_X_y=True)\n",
+    "X = X.reshape((-1, 28, 28))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "_cell_guid": "79c7e3d0-c299-4dcb-8224-4455121ee9b0",
+    "_uuid": "d629ff2d2480ee46fbb7e2d37f6b5fab8052498a"
+   },
+   "outputs": [],
+   "source": [
+    "# For a full-blown example, you can set \n",
+    "# n_train, n_test = 60000, 10000\n",
+    "n_train, n_test = 60, 10\n",
+    "\n",
+    "X_train = X[:n_train]\n",
+    "y_train = y[:n_train]\n",
+    "X_test = X[n_train:n_train+n_test]\n",
+    "y_test = y[n_train:n_train+n_test]\n",
+    "\n",
+    "print(X_train.shape, y_train.shape, X_test.shape, y_test.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Some examples of the input data\n",
+    "We choose the first 20 samples from the training set and visualize them."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_heatmap(X_train[0])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_heatmap(X_train[1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Binarization of the images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "binarizer = Binarizer(threshold=0.4)\n",
+    "X_train_binarized = binarizer.fit_transform(X_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# NOTE TO UMBE: plot_heatmap does not support binary arrays\n",
```

Comment:
Noted. We should be able to easily change this.

Reply:
I created #444

Reply:
I will wait until it's merged and update the notebook.

Reply:
Thanks. May I regard that PR as approved?